### PR TITLE
[Spi_host/doc] updated documentation to reach V1 ready

### PR DESCRIPTION
**Pull request by rasmus-madsen**
This PR includes 3 commits.
1. Update test plan with test points for both V1 and V2
    Update Coverage plan with functional coverage points
    several test points were added - and the original ones are cleaned up. 
    I also removed the list of tests that were mentioned in each test point as none of these exist and therefore should not be listed 
2. update dv plan so that should now be ready for V1 review
3. updated checklist to track the status of the documentation.

I also added the test plan to the hardware dashboard.

A PR will follow this with the Spi host environment and smoke test.

with approval of this document, I will consider the SPI_HOST documentation reviewed and approved for V1.
IP will stay at V0 until the checklist is complete and reviewed.

### CODE REVIEW

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 1]
```diff
@@ -18,12 +18,7 @@
 
             Stimulus:
               - Enable spi_host ip
-              - Clear/enable interrupt (if needed)
-              - Program configuration registers (CONTROL) to reset spi_fsm, reset/set watermark for the fifos
-              - Program the CONFIGOPTS register to config spi_host channel
-                operating in different modes (single, dual, or quad)
-              - Randomize the content of COMMAND register
-              - Program/retrive data in TXDATA/RXDATA register w.r.t transmitted/received transactions
+              - Write data in standard mode - and read it back
```
Comment (rasmus-madsen) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:21]: the smoke test should be simple - the original list was too much to cover in a simple V1 test IMO

File: hw/ip/spi_host/doc/checklist.md [hunk 2]
```diff
@@ -125,17 +125,17 @@ Testbench     | [CSR_CHECK_GEN_AUTOMATED][]           | Not Started |
 Testbench     | [TB_GEN_AUTOMATED][]                  | Not Started |
 Tests         | [SIM_SMOKE_TEST_PASSING][]            | Not Started |
 Tests         | [SIM_CSR_MEM_TEST_SUITE_PASSING][]    | Not Started |
-Tests         | [FPV_MAIN_ASSERTIONS_PROVEN][]        | Not Started |
+Tests         | [FPV_MAIN_ASSERTIONS_PROVEN][]        | DONE        |
```
Comment (weicaiyang) [inline hw/ip/spi_host/doc/checklist.md:128]: we don't have a standalone FPV TB for it, so it's N/A
```suggestion
Tests         | [FPV_MAIN_ASSERTIONS_PROVEN][]        | N/A        |
```

File: hw/ip/spi_host/doc/checklist.md [hunk 3]
```diff
@@ -125,17 +125,17 @@ Testbench     | [CSR_CHECK_GEN_AUTOMATED][]           | Not Started |
...
 Tool Setup    | [SIM_ALT_TOOL_SETUP][]                | Not Started |
 Regression    | [SIM_SMOKE_REGRESSION_SETUP][]        | Not Started |
 Regression    | [SIM_NIGHTLY_REGRESSION_SETUP][]      | Not Started |
-Regression    | [FPV_REGRESSION_SETUP][]              | Not Started |
+Regression    | [FPV_REGRESSION_SETUP][]              | N/A         |
```
Comment (weicaiyang) [inline hw/ip/spi_host/doc/checklist.md:132]: we should add it to FPV regression to check the assertion in RTL. https://reports.opentitan.org/hw/top_earlgrey/formal/summary.html
Perhaps @cindychip can help on this

File: hw/ip/spi_host/doc/checklist.md [hunk 4]
```diff
@@ -125,17 +125,17 @@ Testbench     | [CSR_CHECK_GEN_AUTOMATED][]           | Not Started |
...
 Coverage      | [SIM_COVERAGE_MODEL_ADDED][]          | Not Started |
 Code Quality  | [TB_LINT_SETUP][]                     | Not Started |
-Integration   | [PRE_VERIFIED_SUB_MODULES_V1][]       | Not Started |
-Review        | [DESIGN_SPEC_REVIEWED][]              | Not Started |
-Review        | [TESTPLAN_REVIEWED][]                 | Not Started |
-Review        | [STD_TEST_CATEGORIES_PLANNED][]       | Not Started | Exception (?)
+Integration   | [PRE_VERIFIED_SUB_MODULES_V1][]       | N/A         |
+Review        | [DESIGN_SPEC_REVIEWED][]              | DONE        |
+Review        | [TESTPLAN_REVIEWED][]                 | DONE        |
```
Comment (weicaiyang) [inline hw/ip/spi_host/doc/checklist.md:137]: not sure if we have review this in a meeting?

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 5]
```diff
@@ -144,7 +211,58 @@
...
+            both scenarios with both a slow spi_sck, a very fast spi_sck (2x core_clk)
+            and where the two clocks are very close has been tested
+            '''
+    }
+    {
+      name: unaligned_data_cg
+      desc: '''
+            Collect coverage the alignment of writes to the data window
+            to verify that all possible alignments was seen
+            '''
```
Comment (weicaiyang) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:265]: based on the testplan, there are more coverpoints we can add, such as speed(single, dual, quad), pass through, clock divider (max/min frequency), endian setting, duplex(half and full ) etc

File: hw/ip/spi_host/doc/checklist.md (hunk 4 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/doc/checklist.md:137]: @weicaiyang no review yet.  'I was hoping this PR could serve as the review. not sure we need a meeting.

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 5 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:265]: sure my assumptions was any register setting is a given..
but let me add them all  to make it clear

File: hw/ip/spi_host/doc/checklist.md (hunk 4 above)
Comment (weicaiyang) [reply hw/ip/spi_host/doc/checklist.md:137]: usually we do need to meeting to go over it with designer and SW people unless it's a very small block and all owners agree to waive it.

File: hw/ip/spi_host/doc/checklist.md (hunk 2 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/doc/checklist.md:128]: oh shoot. too much copy paste - fixed

File: hw/ip/spi_host/doc/checklist.md (hunk 3 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/doc/checklist.md:132]: ok I will change it back to not started.
@cindychip can you help me get this in?

**Comment by rasmus-madsen** [comment]
@weicaiyang thank you for your feedback I have updated to follow your comments.
@sriyerg do we need a formal doc review or is it ok to update the checklist for doc_reviewed?

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 1 above)
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:21]: Is this just abbreviating the steps on the left?

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 6]
```diff
@@ -66,68 +61,140 @@
...
-      tests: ["spi_host_component_reset"]
+      tests: []
     }
     {
-      name: clock_domain
+      name: modes
       desc: '''
-            TBD - Verify the function of clock-crossing domain for spi_host
+            - Verify the function of spi_host in different special_modes
+              (pass-through and manual_ctrl_cs)
```
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:128]: Manual CS control has been removed from the spec.

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 7]
```diff
@@ -66,68 +61,140 @@
                 once the event interrupt pin is asserted
             '''
       milestone: V2
-      tests: ["spi_host_error_event_intr"]
```
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:69]: So are these removed directed tests now effectively handled as covergroups in the new plan?

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 8]
```diff
@@ -66,68 +61,140 @@
...
+      tests: []
     }
     {
-      name: tx_overflow_error
+      name: clock_rate
       desc: '''
-            Test TX OVERFLOW error is asserted by the spi_host
+            Stimulus:
+              - assert CONFIGOPTS_0.CSAAT_0 to combine messages
```
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:70]: Note: CSAAT is now a COMMAND field.

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 9]
```diff
@@ -45,7 +40,7 @@
               - Ensure transactions are transmitted/received correctly
             '''
       milestone: V2
-      tests: ["spi_host_perf"]
+      tests: []
```
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:43]: Perhaps this is a dumb question, but what does it mean if there is a test point with no tests specified?  (I feel like I should know this already...)

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 10]
```diff
@@ -66,68 +61,140 @@
...
     }
     {
       name: component_reset
       desc: '''
             Test components (spi_fsm, rx_fifo, tx_fifo) are randomly reset
 
             Stimulus:
-              - Reset the components of spi_host randomly after a random number of data shows up on fifos
+              - Reset the components of spi_host randomly
```
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:113]: Note: There is now only one sw_rst register for all components.

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 11]
```diff
@@ -66,68 +61,140 @@
...
-              - TBD
-      '''
+              - Check that the data can be read one full cycle after the data was asserted
+            '''
       milestone: V2
-      tests: ["spi_host_special_modes"]
+      tests: []
+    }
+    {
+      name: endian
```
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:163]: This looks like a reasonable test, though I was under the impression that it was not possible to test different parameter values, for reasons very specific to our flow?

I think it would be good to document that this is a test that should be done *in principle* but I imagine that we would get absolutely zero resistance to waiving it, as OpenTitan will never use it.   (It was a feature that was added with an eye to reuse, not to OpenTitan minimum requirements.)  It might be even worthwhile to mark it as low-priority in the testplan, as this is so low priority that the only reason we should put it in the test plan is to document that it is a designed but unexplored feature.

Comment (martin-lueker) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:163]: The same comments apply to multiple CS lines.  They are not going to be used in the chip, so we should perhaps consider documenting them as features we have no immediate plans of testing.

File: hw/ip/spi_host/data/spi_host_testplan.hjson [hunk 12]
```diff
@@ -66,68 +61,140 @@
...
+              - in standard mode set the DUT to run full duplex
+
+            Checking:
+              - verify that the DUT support both half and full duplex in standard mode.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tx_only
```
Comment (martin-lueker) [inline hw/ip/spi_host/data/spi_host_testplan.hjson:188]: I think we need something for rx-only segments as well (to ensure that tx data isn't consumed during those)

**Review by martin-lueker — APPROVED** [review]

**Review by martin-lueker — CHANGES_REQUESTED** [review]
Some comments about various older features being removed, as well questions about our general approach in testing Parameter Values that are not immediately needed in this TapeOut.

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 6 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:128]: Thank I have removed it

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 1 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:21]: @martin-lueker it is simplifying them.
the first test should just be very basic.  So for V1 this will only support standard mode an a minimum of randomization,

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 9 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:43]: just means the test that exercises this test point has not been implemented yet.
I removed the tests for two reasons
1. the build system will give you warnings for all the tests defined in the hjson but not yet implemented.
2. I don't plan on making a test for each test point. most will be covered by one or two tests.

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 10 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:113]: fixed - I updated the wording

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 11 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:163]: @martin-lueker I thought about that - but I am unsure about how to capture "planned but waived" as I think it is important information.
@weicaiyang @sriyerg any comments on how to track deferred testscases?


for multiple cs_lines I think that as long as it is listed as a supported feature I think we should test it.

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 12 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:188]: @martin-lueker at first I was thinking that it would not be needed.
but on second thought I think it is a good idea. and certainly can hurt. I have updated the test point to include both

File: hw/ip/spi_host/doc/checklist.md (hunk 4 above)
Comment (rasmus-madsen) [reply hw/ip/spi_host/doc/checklist.md:137]: ok I have reset to not started.. I don't think this is a major issue for me.

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 11 above)
Comment (weicaiyang) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:161]: Agree with Rasmus, it seems weird that our spec says we support a feature, but it's not verified. In my previous experience, if some features aren't verified, we will document those aren't supported and user can use those with their own risk.

Comment (martin-lueker) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:161]: Okay, well, perhaps I can just enumerate them as implemented, but not supported/not verified.

Comment (martin-lueker) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:161]: Documentation to be changed as part of #7553

**Review by martin-lueker — APPROVED** [review]
LGTM Thanks Rasmus

**Comment by sriyerg** [comment]
All testplan reviews (meeting) so far have proven to be useful. Recently, the SW team has also requested to have more visibility into testplan reviews so that they have a better idea of what functionality is planned to be tested and what is expected to be working at the full chip at each stage. Please do schedule one, thanks!

File: hw/ip/spi_host/data/spi_host_testplan.hjson (hunk 11 above)
Comment (sriyerg) [reply hw/ip/spi_host/data/spi_host_testplan.hjson:161]: If the design is parameterized (i.e. it does not refer to the parameters from some `foo_pkg`), then we need to parameterize the UVM testbench as well, so that its easy to get other 'flavors' verified later when they get used in a different OT implementation.

**Comment by rasmus-madsen** [comment]
> All testplan reviews (meeting) so far have proven to be useful. Recently, the SW team has also requested to have more visibility into testplan reviews so that they have a better idea of what functionality is planned to be tested and what is expected to be working at the full chip at each stage. Please do schedule one, thanks!

@sriyerg I was never trying to avoid a review - just to get the doc V1 approved.
but as per agreement with @weicaiyang this was also reset to not started.
so I think we are all in sync!

**Comment by sriyerg** [comment]
> > All testplan reviews (meeting) so far have proven to be useful. Recently, the SW team has also requested to have more visibility into testplan reviews so that they have a better idea of what functionality is planned to be tested and what is expected to be working at the full chip at each stage. Please do schedule one, thanks!
> 
> @sriyerg I was never trying to avoid a review - just to get the doc V1 approved.
> but as per agreement with @weicaiyang this was also reset to not started.
> so I think we are all in sync!

Great, thanks!
